### PR TITLE
Embedding request factory

### PR DESCRIPTION
**Pull request by woksin**
Adds specs to and implements the EmbeddingRequestFactory

### CODE REVIEW

File: Specifications/Embeddings.Processing/for_EmbeddingRequestFactory/when_creating/a_delete_request/and_everything_works.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using Dolittle.Runtime.Events.Processing.Projections;
+using Dolittle.Runtime.Protobuf;
+using Machine.Specifications;
+using It = Machine.Specifications.It;
+
+namespace Dolittle.Runtime.Embeddings.Processing.for_EmbeddingRequestFactory.when_creating.a_delete_request
+{
+    public class and_everything_works : given.all_dependencies
+    {
+        static UncommittedEvent @event;
```
Comment (joelhoisko) [inline Specifications/Embeddings.Processing/for_EmbeddingRequestFactory/when_creating/a_delete_request/and_everything_works.cs:15]:
```suggestion
```

Like the action says, unused field.

File: Specifications/Embeddings.Processing/for_EmbeddingRequestFactory/when_trying_to_create/a_compare_request/and_everything_works.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using Dolittle.Runtime.Events.Store;
+using Dolittle.Runtime.Rudimentary;
+using Machine.Specifications;
+using It = Machine.Specifications.It;
+
+namespace Dolittle.Runtime.Embeddings.Processing.for_EmbeddingRequestFactory.when_trying_to_create.a_compare_request
+{
+    public class and_everything_works : given.all_dependencies
+    {
+        static UncommittedEvent @event;
```
Comment (joelhoisko) [inline Specifications/Embeddings.Processing/for_EmbeddingRequestFactory/when_trying_to_create/a_compare_request/and_everything_works.cs:13]:
```suggestion
```

Same thing.

File: Specifications/Embeddings.Processing/for_EmbeddingRequestFactory/when_trying_to_create/a_delete_request/and_everything_works.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+using Dolittle.Runtime.Protobuf;
+using Machine.Specifications;
+using It = Machine.Specifications.It;
+using Dolittle.Runtime.Rudimentary;
+
+namespace Dolittle.Runtime.Embeddings.Processing.for_EmbeddingRequestFactory.when_trying_to_create.a_delete_request
+{
+    public class and_everything_works : given.all_dependencies
+    {
+        static UncommittedEvent @event;
```
Comment (joelhoisko) [inline Specifications/Embeddings.Processing/for_EmbeddingRequestFactory/when_trying_to_create/a_delete_request/and_everything_works.cs:16]:
```suggestion
```

**Review by joelhoisko — CHANGES_REQUESTED** [review]
Neato stuff, the GH Action picked up some warnings so made suggestions on some of them, not sure about that one warning regarding Enumerable methods on indexable collections.

**Comment by woksin** [comment]
Committed your changes. The warning you're referring to we can ignore

**Review by joelhoisko — APPROVED** [review]

**Pull request merged**